raiden/raiden_service.py

```diff
@@ -306,6 +306,20 @@ def register_channel_manager(self, channel_manager):
                 self.config['reveal_timeout'],
             )
 
+    def log_and_apply(self, state_change):
+        self.transact_log.log(state_change)
+
+        messages = self.state_manager.dispatch(state_change)
+
+        for m in messages:
```

Comment:
this may also be an event from the state_manager, used to inform when a mediated transfer is finalized (either with a cancel/failure or success)